ideascube/configuration/models.py

```diff
@@ -0,0 +1,76 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+from .errors import (
+    InvalidConfigurationValue,
+    InvalidStoredConfigurationValue,
+    NoSuchConfigurationKey,
+    NoSuchConfigurationNamespace,
+    )
+from .registry import REGISTRY
+
+
+class Configuration(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @staticmethod
+    def _get_key_registry(namespace, key):
+        try:
+            namespace_registry = REGISTRY[namespace]
+
+        except KeyError:
+            raise NoSuchConfigurationNamespace(namespace)
+
+        try:
+            return namespace_registry[key]
+
+        except KeyError:
+            raise NoSuchConfigurationKey(namespace, key)
+
+    @classmethod
+    def get(cls, namespace, key):
+        key_registry = cls._get_key_registry(namespace, key)
+        expected_type = key_registry['python_type']
+
+        config = cls.objects.get(namespace=namespace, key=key)
+
+        if not isinstance(config.value, expected_type):
+            raise InvalidStoredConfigurationValue(
+                config, expected_type, type(config.value))
+
+        return config
+
+    @classmethod
+    def get_value(cls, namespace, key):
+        try:
+            return cls.get(namespace, key).value
+
+        except cls.DoesNotExist:
+            return REGISTRY[namespace][key]['default']
```

Comment:
The difference between get and get_value is not obvious from the name only.
One will need to read the code to get the behavior. 

Python dict.get return the default value if the key is not present. Here this is get_value who return the default. Maybe we may change the API somehow (the name, make one function private) ?
At least a docstring.